runtime/bcutil/ClassFileOracle.cpp

```diff
@@ -1158,20 +1163,70 @@ ClassFileOracle::walkMethodCodeAttributeAttributes(U_16 methodIndex)
+							/* throw error */
+							PORT_ACCESS_FROM_PORT(_context->portLibrary());
+							_context->javaVM()->dynamicLoadBuffers->classFileError = (U_8 *) j9mem_allocate_memory(sizeof(J9CfrError), J9MEM_CATEGORY_CLASSES);
+							buildError((J9CfrError*)_context->javaVM()->dynamicLoadBuffers->classFileError, J9NLS_CFR_LVTT_INDEX_OUTOFRANGE__ID, GenericErrorCustom, lvttEntry.index);
+							_buildResult = GenericErrorCustom;
 							break;
-						} else if (NULL == _methodsInfo[methodIndex].localVariablesInfo[index].localVariableTypeTable) {
-							_methodsInfo[methodIndex].localVariablesInfo[index].localVariableTypeTable = localVariableTypeTable;
-						} else if (localVariableTypeTable != _methodsInfo[methodIndex].localVariablesInfo[index].localVariableTypeTable) {
+						} else if (NULL == _methodsInfo[methodIndex].localVariablesInfo[lvttEntry.index].localVariableTypeTable) {
+							_methodsInfo[methodIndex].localVariablesInfo[lvttEntry.index].localVariableTypeTable = localVariableTypeTable;
+						} else if (localVariableTypeTable != _methodsInfo[methodIndex].localVariablesInfo[lvttEntry.index].localVariableTypeTable) {
 							Trc_BCU_ClassFileOracle_walkMethodCodeAttributeAttributes_DuplicateLocalVariableTypeTable(
-									(U_32)getUTF8Length(_classFile->methods[methodIndex].nameIndex), getUTF8Data(_classFile->methods[methodIndex].nameIndex), localVariableTypeTable, _methodsInfo[methodIndex].localVariablesInfo[index].localVariableTypeTable);
+									(U_32)getUTF8Length(_classFile->methods[methodIndex].nameIndex), getUTF8Data(_classFile->methods[methodIndex].nameIndex), localVariableTypeTable, _methodsInfo[methodIndex].localVariablesInfo[lvttEntry.index].localVariableTypeTable);
 							_buildResult = GenericError;
 							break;
 						}
+
+						/* Verify that a matching entry exists in a LocalVariableTable entry. */
+						if  (NULL != _methodsInfo[methodIndex].localVariablesInfo[lvttEntry.index].localVariableTable) {
+							/* Check to see if an entry at the lvtt index has been recorded. This should be the common case.
+							 * Otherwise we will need to check each of the code attributes attribute table entries. */
+							J9CfrLocalVariableTableEntry* lvtEntries = _methodsInfo[methodIndex].localVariablesInfo[lvttEntry.index].localVariableTable->localVariableTable;
+							U_16 lvtLength = _methodsInfo[methodIndex].localVariablesInfo[lvttEntry.index].localVariableTable->localVariableTableLength;
+							for (U_16 localVariableTableIndex = 0;  localVariableTableIndex < lvtLength; localVariableTableIndex++) {
+								if ((lvttEntry.index == lvtEntries[localVariableTableIndex].index)
+									&& (lvttEntry.startPC== lvtEntries[localVariableTableIndex].startPC)
+									&& (lvttEntry.length == lvtEntries[localVariableTableIndex].length)
+									&& (lvttEntry.nameIndex== lvtEntries[localVariableTableIndex].nameIndex)) {
```

Comment:
Please change to
&& (lvttEntry.nameIndex== lvtEntries[localVariableTableIndex].nameIndex)
) {